src/modules/jj.rs

```diff
@@ -0,0 +1,290 @@
+use super::{Context, Module, ModuleConfig};
+use crate::configs::jj::JJConfig;
+use crate::formatter::string_formatter::StringFormatterError;
+use crate::formatter::StringFormatter;
+use std::borrow::Cow;
+
+/// Creates a module with the jj status in the current directory
+///
+/// Will display the current branch, commit and change id of the jj repository in the current directory
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("jj_status");
+    let config = JJConfig::try_load(module.config);
+    // check if the current directory is a jj repo
+    let repo_root = context
+        .exec_cmd("jj", &["root", "--ignore-working-copy"])?
+        .stdout;
+
+    let template = jj_template(config.truncation_length);
+    let output = context.exec_cmd(
+        "jj",
+        &[
+            "log",
+            "-r",
+            "@",
+            "-T",
+            template.as_str(),
+            "--no-graph",
+            "--ignore-working-copy",
+            "--no-pager",
+            "--color",
+            "never",
+            "-R",
+            repo_root.trim(),
+        ],
+    )?;
+    let parsed = JJStatus::new(&output.stdout)?;
+    let formatter = match StringFormatter::new(config.format) {
+        Ok(formatter) => formatter,
+        Err(e) => {
+            log::error!("Error in parsing format string for jj module {}", e);
+            return None;
+        }
+    };
+    let formatted = formatter
+        .map_meta(|s, _| match s {
+            "symbol" => Some(config.symbol),
+            "divergent_symbol" => {
+                if parsed.divergent == "true" {
+                    Some(config.divergent_symbol)
+                } else {
+                    None
+                }
+            }
```

Comment:
I suggest to replace this block with
```suggestion
           "divergent_symbol" => parsed
                .divergent
                .eq("true")
                .then_some(config.divergent_symbol),
```
to keep it more consistent with the comparison below